GCSViews/FlightPlanner.cs

```diff
@@ -8004,5 +8006,119 @@ private void gDALOpacityToolStripMenuItem_Click(object sender, EventArgs e)
             if (InputBox.Show("Opacity 0.0-1.0", "Enter opacity (0.0-1.0)", ref ans) == DialogResult.OK)
                 GDAL.GDALProvider.Instance.opacity = double.Parse(InputBox.value);
         }
+
+        private static bool stopInjectCustomMap = false;
+        private void BUT_InjectCustomMap_Click(object sender, EventArgs e)
+        {
+            var map = new GMapControl();
+            var tilesCount = new Dictionary<int, int>();
+            try
+            {
+                if (BUT_InjectCustomMap.Text == Strings.Cancel)
+                {
+                    stopInjectCustomMap = true;
+                    return;
+                }
+                stopInjectCustomMap = false;
+
+                map.MapProvider = GoogleSatelliteMapProvider.Instance;
+
+                map.CacheLocation = Settings.GetDataDirectory() +
+                                        "gmapcache" + Path.DirectorySeparatorChar;
+
+                var fbd = new FolderBrowserDialog();
+                fbd.SelectedPath = @"C:\";
+
+                if (fbd.ShowDialog() != DialogResult.OK)
+                {
+                    map.Dispose();
+                    return;
+                }
+
+                if (fbd.SelectedPath != "")
+                {
+                    BUT_InjectCustomMap.Text = Strings.Cancel;
+                    progressBarInjectCustomMap.Value = 0;
+                    progressBarInjectCustomMap.Visible = true;
+
+                    var files_jpg = Directory.GetFiles(fbd.SelectedPath, "*.jpg", SearchOption.AllDirectories);
+                    var files_jpeg = Directory.GetFiles(fbd.SelectedPath, "*.jpeg", SearchOption.AllDirectories);
+                    var files_png = Directory.GetFiles(fbd.SelectedPath, "*.png", SearchOption.AllDirectories);
+                    string[] files = new string[files_jpg.Length + files_jpeg.Length + files_png.Length];
+                    Array.Copy(files_jpg, files, files_jpg.Length);
+                    Array.Copy(files_jpeg, files, files_jpeg.Length);

```

Comment:
this is still an issue.   where is the destination index?

Reply:
You're absolutely right ! I'm on it.